lib/sqli/sqli_lexer.re2c

```diff
@@ -182,6 +182,33 @@ sqli_get_token(
           YYSETSTATE(-1);
           RET_DATA(DATA, ctx, arg);
       }
+      <BITVALUE> [0-1] {
+          if (!detect_buf_add_char(&ctx->lexer.buf, DETECT_RE2C_YYCURSOR(&ctx->lexer.re2c)[-1])) {
+              DETECT_RE2C_UNUSED_BEFORE(&ctx->lexer.re2c);
+              goto sqli_BITVALUE;
+          } else {
+              detect_buf_deinit(&ctx->lexer.buf);
+              RET(ctx, TOK_ERROR);
+          }
+      }
+      <QBITVALUE> [0-1] {
+          if (!detect_buf_add_char(&ctx->lexer.buf, DETECT_RE2C_YYCURSOR(&ctx->lexer.re2c)[-1])) {
+              DETECT_RE2C_UNUSED_BEFORE(&ctx->lexer.re2c);
+              goto sqli_QBITVALUE;
+          } else {
+              detect_buf_deinit(&ctx->lexer.buf);
+              RET(ctx, TOK_ERROR);
+          }
```

Comment:
`<BITVALUE> [0-1]` and `<QBITVALUE> [0-1]` can be combined into a single rule. `goto yy0` should be used to do that. Se how it works for quoted characters:
```
      <INSTRING,SQUOTE,DQUOTE,BQUOTE> . {
          detect_buf_add_char(&ctx->lexer.buf, DETECT_RE2C_YYCURSOR(&ctx->lexer.re2c)[-1]);
          DETECT_RE2C_UNUSED_BEFORE(&ctx->lexer.re2c);
          goto yy0;
      }
```